File: base/carro/.cache/cheat/ts/adapter.ts
import {Car} from './car';
class Adapter {
    car: Car;

    constructor() {
        this.car = new Car();
    }

    enter(): void {
        this.car.enter();
    }

    leave(): void {
        this.car.leave();
    }

    fuel(gas: number): void {
        this.car.fuel(gas);
    }

    drive(km: number): void {
        this.car.drive(km);
    }

    toString(): string {
        return this.car.toString();
    }
}
export {Adapter};

if (require.main === module) {
    console.log("Rodando a adapter...");
}

File: base/carro/.cache/cheat/java/Adapter.java

class Adapter {
    Car car;

    public Adapter() {
        this.car = new Car();
    }
    
    public void enter() {
        car.enter();
    }

    public void leave() {
        car.leave();
    }

    public void fuel(int gas) {
        car.fuel(gas);
    }

    public void drive(int km) {
        car.drive(km);
    }

    public String toString() {
        return car.toString();
    }

    public static void main(String[] args) {
        System.out.println("Rodando adapter");
    }
}